client/webserver/types.go

```diff
@@ -67,3 +75,409 @@ type withdrawForm struct {
+	var v uint64
+	add := func(rate, qty uint64) {
+		if ord.Sell {
+			v += calc.BaseToQuote(rate, qty)
+		} else {
+			v += qty
+		}
+	}
+	for _, match := range ord.Matches {
+		if filter(match) {
+			add(match.Rate, match.Qty)
+		}
+	}
+	return v
+}
+
+// hasLiveMatches will be true if there are any matches < MakerRedeemed.
+func (ord *orderReader) hasLiveMatches() bool {
+	for _, match := range ord.Matches {
+		if match.Status < order.MakerRedeemed {
+			return true
+		}
+	}
+	return false
+}
+
+// StatusString is the order status.
+func (ord *orderReader) StatusString() string {
+	isLive := ord.hasLiveMatches()
+	switch ord.Status {
+	case order.OrderStatusUnknown:
+		return "unknown"
+	case order.OrderStatusEpoch:
+		return "epoch"
+	case order.OrderStatusBooked:
+		if ord.Cancelling {
+			return "cancelling"
+		}
+		return "booked"
+	case order.OrderStatusExecuted:
+		if isLive {
+			return "settling"
+		}
+		return "executed"
+	case order.OrderStatusCanceled:
+		if isLive {
+			return "canceled/settling"
+		}
+		return "canceled"
+	case order.OrderStatusRevoked:
+		if isLive {
+			return "revoked/settling"
+		}
+		return "revoked"
+	}
+	return "unknown"
+}
+
+// RateString is a formatted rate with units.
+func (ord *orderReader) RateString() string {
+	if ord.Type == order.MarketOrderType {
+		return "market"
+	}
+	return fmt.Sprintf("%s %s/%s", precision8(ord.Rate), ord.QuoteSymbol, ord.BaseSymbol)
+}
+
+// MatchReaders creates a slice of *matchReader for the order's matches.
+func (ord *orderReader) MatchReaders() []*matchReader {
+	readers := make([]*matchReader, 0, len(ord.Matches))
+	for _, match := range ord.Matches {
+		readers = append(readers, newMatchReader(match, ord))
+	}
+	return readers
+}
+
+// SwapFeesString is a formatted string of the paid swap fees.
+func (ord *orderReader) SwapFeesString() string {
+	return precision8(ord.FeesPaid.Swap)
+}
+
+// RedemptionFeesString is a formatted string of the paid redemption fees.
+func (ord *orderReader) RedemptionFeesString() string {
+	return precision8(ord.FeesPaid.Redemption)
+}
+
+func (ord *orderReader) FundingCoinIDs() []string {
+	ids := make([]string, 0, len(ord.FundingCoins))
+	for i := range ord.FundingCoins {
+		ids = append(ids, decodeCoinID(ord.FromID(), ord.FundingCoins[i]))
+	}
+	return ids
+}
+
+// matchReader wraps a core.Match and provides methods for info display.
+// Whenever possible, add an matchReader methods rather than a template func.
+type matchReader struct {
+	*core.Match
+	ord *orderReader
+}
+
+func newMatchReader(match *core.Match, ord *orderReader) *matchReader {
+	return &matchReader{
+		Match: match,
+		ord:   ord,
+	}
+}
+
+// StatusString is a formatted string of the match status.
+func (m *matchReader) StatusString() string {
+	switch m.Status {
+	case order.NewlyMatched:
+		return "(0 / 4) Newly Matched"
+	case order.MakerSwapCast:
+		return "(1 / 4) First Swap Sent"
+	case order.TakerSwapCast:
+		return "(2 / 4) Second Swap Sent"
```

Comment:
I feel like these two strings would be better as "Initiator Contract Sent" and "Participant Contract Sent", but I'm ok to leave this until we get other eyes on it.  Part of it is that "swap" is likely to be though of as the entire process.